packages/graph-explorer/src/core/StateProvider/configuration.ts

```diff
@@ -30,35 +31,23 @@ export const configurationAtom = atomWithLocalForage<
   Map<ConfigurationId, RawConfiguration>
 >(new Map(), "configuration");
 
-/** Gets or sets the config that is currently active. */
-export const activeConfigSelector = atom(
-  get => {
-    const configMap = get(configurationAtom);
-    const id = get(activeConfigurationAtom);
-    const activeConfig = id ? configMap.get(id) : null;
-    return activeConfig;
-  },
-  async (get, set, newValue: RawConfiguration | typeof RESET) => {
-    const configId = get(activeConfigurationAtom);
-    if (!configId) {
-      return;
-    }
-    await set(configurationAtom, async prevConfigMap => {
-      const updatedConfigMap = new Map(await prevConfigMap);
-
-      // Handle reset value
-      if (!newValue || newValue === RESET) {
-        updatedConfigMap.delete(configId);
-        return updatedConfigMap;
-      }
-
-      // Update the map
-      updatedConfigMap.set(configId, newValue);
-
-      return updatedConfigMap;
-    });
+/** Gets the currently active config. */
+export const activeConfigSelector = atom(get => {
+  const configMap = get(configurationAtom);
+  const id = get(activeConfigurationAtom);
+  const activeConfig = id ? configMap.get(id) : null;
+  return activeConfig;
+});
+
+export const activeConnectionAtom = atom(get => {
```

Comment:
Created this new atom to replace `activeConnectionSelector` in the `connector.ts` file. It uses the lodash deep equal function to reduce updates to when the value actually changes, similar to the code it replaces.